macauff/matching.py

```diff
@@ -6,8 +6,11 @@
 import os
 import sys
 import warnings
+import datetime
 from configparser import ConfigParser
+from time import sleep
 import numpy as np
+from mpi4py import MPI
```

Comment:
This may be a broader request, and perhaps significant work, but can we make mpi4py an optional dependency? I.e., wrap `from mpi4py import MPI` in a `try` and fall back on serial-only loops if it fails? This way in the broader use-case someone who simply wants to run a small section of sky, or the use of memmapped files in a single very-large block, don't need the additional dependency installed.

Reply:
I've made mpi4py an optional dependency - if it fails to import, `rank` and `comm_size` are set to 0 and 1 respectively so the code falls back to serial mode. A warning is printed if the user has explicitly set the new `use_mpi` flag.